### DISCUSSION TITLE
Performance on Rules

### DISCUSSION THREAD

**Discussion by tsunfinger** [arendst/Tasmota#19676] [Q&A] [answered]
Performance on rule seems to be not very exciting. For example from trigger to action it last 300ms to 2s on same function.
Do I probably overload rules? Can I do something on code to improve performance?

**Comment by sfromis** [comment, marked as answer]
That does not look like about performance, but if you have multiple commands in one rule, the default Backlog inter-command delay is 200 ms, unless you change it, or use `Backlog0` for no delay.

**Reply by tsunfinger** [reply]
200ms per command? I would explain why it takes so long from trigger to buzzer for acoustic confirmation.
